test-cargo-miri/run-test.py

```diff
@@ -70,14 +73,20 @@ def test_no_rebuild(name, cmd):
         fail("Something was being rebuilt when it should not be (or we got no output)");
 
 def test_cargo_miri_run():
+    default_env={
+        'MIRIFLAGS': "-Zmiri-disable-isolation",
+        'MIRITESTVAR': "wrongval", # make sure the build.rs value takes precedence
+    }
     test("`cargo miri run` (no isolation)",
         cargo_miri("run"),
         "run.default.stdout.ref", "run.default.stderr.ref",
         stdin=b'12\n21\n',
-        env={
-            'MIRIFLAGS': "-Zmiri-disable-isolation",
-            'MIRITESTVAR': "wrongval", # make sure the build.rs value takes precedence
-        },
+        env=default_env,
+    )
+    # Special test: run it again *without* `-q` to make sure nothing is being rebuilt (Miri issue #1722)
+    test_no_rebuild("`cargo miri run` (no rebuild, no isolation)",
+        cargo_miri("run", quiet=False) + ["--", ""],
+        env=default_env,
```

Comment:
Why these other changes? Just moving the old command up should do it.

Reply:
> Just moving the old command up should do it.

Actually, no: https://github.com/rust-lang/miri/blob/4ecb5a99e09f9d90e33bc6ce667c592c173175a3/test-cargo-miri/build.rs#L15
But yes, `MIRIFLAGS` can be removed. That's indeed an oversight, sorry.

Reply:
Ah, good catch... I am not very happy about the code duplication here but whatever.

Reply:
> I am not very happy about the code duplication here

Is this better?
```python3
    wrongval = "wrongval"
    test("`cargo miri run` (no isolation)",
        cargo_miri("run"),
        "run.default.stdout.ref", "run.default.stderr.ref",
        stdin=b'12\n21\n',
        env={
            'MIRIFLAGS': "-Zmiri-disable-isolation",
            'MIRITESTVAR': wrongval, # make sure the build.rs value takes precedence
        },
    )
    # Special test: run it again *without* `-q` to make sure nothing is being rebuilt (Miri issue #1722)
    test_no_rebuild("`cargo miri run` (no rebuild)",
        cargo_miri("run", quiet=False) + ["--", ""],
        env={'MIRITESTVAR': wrongval}, # changing the env var causes a rebuild (re-runs build.rs),
                                       # so keep it set
    )
```
I'm not familiar with Python so I don't know if I can deduplicate them further...

Reply:
I meant the duplication between `test_no_rebuild` and `test`... but it's not a big deal, we barely touch this code anyway.

Reply:
Oh... Sorry for misinterpreting the comment... 😰

Reply:
No worries, I was not very clear.